server/resp/src/test/java/org/infinispan/server/resp/JsonCommandsTest.java

```diff
@@ -1200,6 +1203,133 @@ private boolean compareJSONGet(JsonValue result, JsonValue expected, JsonPath...
+      // Test legacy path
+      jp = new JsonPath("$");
+      jv = defaultJsonParser.createJsonValue("[1, \"a\", {\"o\":\"v\"}]");
+      // Append to root
+      redis.jsonSet(key, jp, jv);
+      res = (Long) command.jsonArrinsert(key, ".", 2, "\"aString\"", "1", "{\"aObj\": null }");
+      assertThat(res).isEqualTo(6);
+      jsonGet = redis.jsonGet(key, jp);
+      assertThat(jsonGet.get(0).toString()).isEqualTo("[[1,\"a\",\"aString\",1,{\"aObj\":null},{\"o\":\"v\"}]]");
+      assertThatThrownBy(() -> command.jsonArrinsert("non-existent", "$", 1, "v1", "v2", "v3"))
+            .isInstanceOf(RedisCommandExecutionException.class)
+            .hasMessage("ERR could not perform this operation on a key that doesn't exist");
+
+      jv = defaultJsonParser.createJsonValue("""
+               {"a": 2,
+               "arr_value": ["one", "two", "three"],
+               "nested":
+                  {"a": true,
+                  "nested2": {
+                   "foo": "fore"},
+                   "arr_value": [1,2,3,4]
+                  },
+               "nested1":
+                  {
+                   "arr_value": null
+                  },
+               "nested2":
+                  {
+                   "arr_value": 1
+                  }
+               }
+            """);
+
+      redis.jsonSet(key, jp, jv);
+      // Insert to single element (at the end)
+      jp = new JsonPath(".nested.arr_value");
+      JsonValue v4 = defaultJsonParser.fromObject("aString");
+      JsonValue v5 = defaultJsonParser.fromObject(1);
+      JsonValue v6 = defaultJsonParser.createJsonValue("{\"aObj\": null}");
+      jsonArrappend = redis.jsonArrinsert(key, jp, 4, v4, v5, v6);
+      assertThat(jsonArrappend).containsExactly(7L);
+      jsonGet = redis.jsonGet(key, jp);
+      assertThat(jsonGet.get(0).toString()).isEqualTo("[1,2,3,4,\"aString\",1,{\"aObj\":null}]");
+      // Append to multiple elements
+      jp = new JsonPath("..arr_value");
+      assertThat(redis.jsonArrinsert(key, jp, 1, v4, v5, v6)).containsExactly(10L);
+      jp = new JsonPath(".nested.*");
+      assertThat(redis.jsonArrinsert(key, jp, 1, v4, v5, v6)).containsExactly(13L);
+      JsonPath jp1 = new JsonPath(".nested2.*");
+      assertThatThrownBy(() -> redis.jsonArrinsert(key, jp1, 1, v4, v5, v6))
+            .isInstanceOf(RedisCommandExecutionException.class)
+            .hasMessage("ERR Path '$.nested2.*' does not exist or not an array");
+      JsonPath jp2 = new JsonPath(".non-existent");
+      assertThatThrownBy(() -> redis.jsonArrinsert(key, jp2, 1, v4, v5, v6))
+            .isInstanceOf(RedisCommandExecutionException.class)
+            .hasMessage("ERR Path '$.non-existent' does not exist or not an array");
+      assertThatThrownBy(() -> command.jsonArrinsert("non-existent", "$", 1, "v1", "v2", "v3"))
+            .isInstanceOf(RedisCommandExecutionException.class)
+            .hasMessage("ERR could not perform this operation on a key that doesn't exist");
```

Comment:
I think we should extract this into a constant but this can be done in another PR